Sources_v3/StreamChatUI/Utils/ChatMessage+Extensions.swift

```diff
@@ -0,0 +1,23 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+
+extension _ChatMessage {
+    var isInteractionEnabled: Bool {
+        guard deletedAt == nil else { return false }
+
+        return localState == nil || lastActionFailed
+    }
+    
+    var lastActionFailed: Bool {
+        switch localState {
+        case .sendingFailed, .syncingFailed, .deletingFailed:
+            return deletedAt == nil
+        default:
+            return false
+        }
+    }
```

Comment:
Please add documentation

Reply:
added in 67d163c3f8bb4855f8fa74fbce82dac8ae80bb23